devbox/common/src/Skipper.scala

```diff
@@ -102,36 +105,55 @@ object Skipper{
       }
     }
 
-    def initialScanIsPathSkipped(base: os.Path, path: os.SubPath, isDir: Boolean) = {
-      // During the initial scan, we want to look up the git index and gitignore
-      // files early, rather than waiting for them to turn up in our folder walk,
-      // so we can get an up to date skipper as soon as possible. Otherwise we
-      // might end up walking tons of files we do not care about only to realize
-      // later that they are all ignored
-      if (isDir) updateIgnoreCache(base, path / ".gitignore")
-      if (path == os.sub) updateIndexCache(base, os.sub / ".git" / "index")
-      !checkFileActive(base, path, isDir)
-    }
+    def isGitPath(path: os.SubPath): Boolean =
+      path.segments.lift(0).contains(".git")
+
+    def initialScanIsPathSkipped(base: os.Path, path: os.SubPath, isDir: Boolean): Boolean =
+      (isGitPath(path) && !syncGit) || {
+        // During the initial scan, we want to look up the git index and gitignore
+        // files early, rather than waiting for them to turn up in our folder walk,
+        // so we can get an up to date skipper as soon as possible. Otherwise we
+        // might end up walking tons of files we do not care about only to realize
+        // later that they are all ignored
+        if (isDir) updateIgnoreCache(base, path / ".gitignore")
+        //if (path == os.sub) updateIndexCache(base, os.sub / ".git" / "index")
+        !checkFileActive(base, path, isDir)
+      }
 
     def batchRemoveSkippedPaths(base: os.Path, paths: PathMap[Boolean]) = {
-      for(p <- ignoreNodeCache.keySet){
-        if (!os.isFile(base / p / ".gitignore") && ignoreNodeCache.contains(p)) {
-          ignoreNodeCache.remove(p)
-          isPathIgnoredCache.clear()
+        for (p <- ignoreNodeCache.keySet) {
+          if (!os.isFile(base / p / ".gitignore") && ignoreNodeCache.contains(p)) {
+            ignoreNodeCache.remove(p)
+            isPathIgnoredCache.clear()
+          }
         }
-      }
 
-      for((p0, isDir) <- paths.walkValues() if !isDir){
-        val p = os.SubPath(p0)
-        if (p.last == ".gitignore") updateIgnoreCache(base, p)
-        updateIndexCache(base, p)
```

Comment:
Let's preserve this guy. We might need to duplicate the `forceIncludes` path set and keep it mutable, matching the existing behavior of `gitIndexCache`, so we can continue to update our git index/ignore logic in response to changing `.gitignore` files in the repository